packages/xstate-solid/src/utils.ts

```diff
@@ -0,0 +1,73 @@
+import type { AnyState, AnyStateMachine, State, StateFrom } from 'xstate';
+import type { SetStoreFunction } from 'solid-js/store';
+import { produce, reconcile } from 'solid-js/store';
+import rfdc from 'rfdc';
+
+// List of keys to reconcile while merging state
+const reconcileKeys: Array<keyof AnyState> = ['context', 'value'];
```

Comment:
I wonder if value should be reconciled. Do you have an example for when it makes sense? I wonder if it isn't enough here to only track its top-level changes. Tracking this deeply will introduce some weird-ish inconsistencies with `state.matches` (as that tracks the whole `state.value`)